executor.js

```diff
@@ -98,55 +98,42 @@ async function handleInput(line, codeCaller) {
   };
 }
 
-// Our overall loop looks like this: read a line of input from stdin, spin
-// off a python worker to handle it, and write the results back to stdout.
-const rl = readline.createInterface({
-  input: process.stdin,
-  output: process.stdout,
-  terminal: false,
-});
-
 let questionTimeoutMilliseconds = Number.parseInt(process.env.QUESTION_TIMEOUT_MILLISECONDS);
 if (Number.isNaN(questionTimeoutMilliseconds)) {
   questionTimeoutMilliseconds = 10000;
 }
 
-let codeCaller = new CodeCallerNative({
-  dropPrivileges: true,
-  questionTimeoutMilliseconds,
-  errorLogger: console.error,
-});
-codeCaller.ensureChild();
+(async () => {
+  let codeCaller = new CodeCallerNative({
+    dropPrivileges: true,
+    questionTimeoutMilliseconds,
+    errorLogger: console.error,
+  });
+  await codeCaller.ensureChild();
```

Comment:
Not that it really matters, but under the new scheme could we move `ensureChild()` inside the constructor to make sure we never forget it? Can we even have async constructors?

Reply:
We cannot have async constructors 😢 it would be very nice if we could (that could simplify some of the code caller pool code), but I think this is the best we can do.

Reply:
Yep, I've just been reading about all the hideous hacks that people have used to partially emulate async constructors and I agree we don't want to go there!